src/reporting/clients/IReportsClient.ts

```diff
@@ -0,0 +1,149 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+import type { AccessToken } from "@itwin/core-bentley";
+import type { Report, ReportCreate, ReportMapping, ReportMappingCreate, ReportUpdate } from "../interfaces/Reports";
+import type { EntityListIterator } from "../iterators/EntityListIterator";
+
+export interface IReportsClient{
+  /**
+   * Gets all Reports within the context of a Project. This method returns the full list of reports.
+   * @param {string} projectId The Project Id.
+   * @param {string} accessToken OAuth access token with scope `insights:read`
+   * @param {boolean} deleted parameter to specify whether to include deleted reports
+   * @param {number} top the number of entities to load.
+   * @memberof ReportingClient
+   * @link https://developer.bentley.com/apis/insights/operations/get-project-reports/
+   */
+  getReports(
+    accessToken: AccessToken,
+    projectId: string,
+    deleted: boolean,
```

Comment:
`deleted` should be optional. Also move it after the `top` parameter.

Reply:
Same applies for `getReportsIterator`.

Reply:
swapped and made deleted optional